cyclonedx/model/component.py

```diff
@@ -107,7 +107,7 @@ def __init__(self, name: str, component_type: ComponentType = ComponentType.LIBR
                  external_references: Optional[List[ExternalReference]] = None,
                  properties: Optional[List[Property]] = None, release_notes: Optional[ReleaseNotes] = None,
                  # Deprecated parameters kept for backwards compatibility
-                 namespace: Optional[str] = None, license_str: Optional[str] = None
+                 namespace: Optional[str] = None, license_str: Optional[str] = None, cpe: Optional[str] = None
```

Comment:
@madpah the depricated parametes on the bottom ... this makes it hard to add additional non-depricated parametes after them.
how do you suggest to add new parameters? should they be added before the depricateds? 
we could use  [PEP 3102](https://www.python.org/dev/peps/pep-3102/) to force them as keyword-arguments.

Reply:
@jkowalleck - I would have instinctively added `cpe` just before the deprecated parameters.

However, PEP-3102 looks to be of interest here as you point out. I assume you are suggesting the function signature be changed to something like:

```python
 ...
properties: Optional[List[Property]] = None, release_notes: Optional[ReleaseNotes] = None,
cpe: Optional[str] = None, *,
# Deprecated parameters kept for backwards compatibility
namespace: Optional[str] = None, license_str: Optional[str] = None
...
```

Reply:
Raised https://github.com/CycloneDX/cyclonedx-python-lib/issues/143 to cover this later.